faq/openfabrics.inc

```diff
@@ -2252,73 +2242,53 @@ shell$ mpirun --mca coll_fca_enable 1 --mca coll_fca_np &lt;N&gt; ...
 /////////////////////////////////////////////////////////////////////////
 $q[] = "Does Open MPI support <code>MXM</code>?";
 $anchor[] = "mxm";
+$a[] = "MXM support is currently deprecated and replaced by <a href=\"#ucx\">UCX</a>.";
+
+/////////////////////////////////////////////////////////////////////////
+$q[] = "Does Open MPI support <code>UCX</code>?";
+$anchor[] = "ucx";
 $a[] = "Yes.
 
-MXM is a <strong>M</strong>ellano<strong>X</strong> <strong>M</strong>essaging
-library which provides enhancements to parallel communication libraries.</br>
+UCX is an open-source optimized communication library which supports multiple
+networks, including RoCE, InfiniBand, uGNI, TCP, shared memory, and others. UCX
+mixes-and-matches transports and protocols which are available on the system
+to provide optimal performance. It also has built-in support for GPU transports
+(with Cuda and RoCM providers) which lets RDMA-capable transports access the
+GPU memory directly.
 
-It is integrated into the Mellanox OFED stack, and also provided in
-a stand-alone package.</br>
+UCX can be downloaded from:
+ <a href=\"http://www.openucx.org\" target=\"&#95;blank\">http://www.openucx.org</a>
 
-The MXM stand-alone package is available for download here: "
-. "<a href=\"http://www.mellanox.com/products/mxm\" target=\"&#95;blank\">" .
-"http://www.mellanox.com/products/mxm</a>
 
-*MX>M requirements*
-<ul>
-<li>
-Mellanox OFED 1.5.3 or later
-</li>
-<li>
-Open MPI 1.6 or later
-</li>
-</ul>
+*Building Open MPI with UCX support*
 
-*Building Open MPI with MXM support*
-
-By default, MXM is installed in [/opt/mellanox/mxm]. Use the following
-configure option to enable MXM support in OMPI:
+Use the following configure option to enable UCX support in OMPI. It will search
+for UCX headers and libraries in the default installation prefix, which is /usr:
 <geshi bash>
-shell$ ./configure --with-mxm=/opt/mellanox/mxm ...
+shell$ ./configure --with-ucx ...
 </geshi>
 
-*Enabling MXM in Open MPI 1.6.1 or later*
-
-In Open MPI 1.6 and above, MXM will be enabled by default, regardless of the number of ranks.
-
-Running an Open MPI job with MXM:
-
-MXM can be run in one of two ways:
+*Running with UCX*
 
-<ul>
-<li>
-With the CM PML and the MXM MTL:
+UCX is enabled and selected by default, so no additional parameters are required.
+In this case the network port with the highest bandwidth on the system will be
+used for inter-node communication, and shared memory will be used intra-node
+communication. To select a specific network device to use (for example, mlx5_0 device port 1):
 <geshi bash>
-shell$ mpirun --mca pml cm --mca mtl mxm ...
+shell$ mpirun -x UCX_NET_DEVICES=mlx5_0:1 ...
 </geshi>
-</li>
-<li>
-With the Yalla PML:
+
+It's also possible to force using UCX for MPI point-to-point and one-sided operations:
 <geshi bash>
-shell$ mpirun --mca pml yalla ...
+shell$ mpirun -mca pml ucx -mca osc ucx ...
```

Comment:
2 dashes before `mca` is preferred (even though it works with both single and double dash).